Branch-SDK/Branch-SDK/BranchUniversalObject.m

```diff
@@ -0,0 +1,137 @@
+//
+//  BranchUniversalObject.m
+//  Branch-TestBed
+//
+//  Created by Derrick Staten on 10/16/15.
+//  Copyright © 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BranchUniversalObject.h"
+#import "BNCError.h"
+#import "BranchConstants.h"
+
+@implementation BranchUniversalObject
+
+- (instancetype)initWithCanonicalIdentifier:(NSString *)canonicalIdentifier {
+    if (self = [super init]) {
+        self.canonicalIdentifier = canonicalIdentifier;
+    }
+    return self;
+}
+
+- (instancetype)initWithTitle:(NSString *)title {
+    if (self = [super init]) {
+        self.title = title;
+    }
+    return self;
+}
+
+- (void)registerView {
+    Branch *branch = [Branch getCurrentInstanceIfAny];
+    if (!branch) {
```

Comment:
All of our other methods will automatically init branch if its not done so by the developer. I think we should take that same course here and create the singleton automatically for the developer. You already have the logic to call in it automatically anyways.